### PR TITLE
overview: allow editing motd

### LINKED ISSUE #8296: Allow the motd to be set from the System page
The system page of Cockpit displays the current contents of /etc/motd and allows the user to dismiss it until it has changed.  Allow the same widget to edit motd if the logged in user is an administrator.  This will probably involve adding some sort of edit button to the page on or near motd widget.  There will need to be a way to edit after the banner has been dismissed so it may need to be near instead of in the banner display widget.

### PR DESCRIPTION
**Pull request by KKoukiou**
For release notes:

Overview: introduce editing of /etc/motd
----------------------------------------------------

Screen Shot 2020-12-11 at 14 38 24

Closes #8296

### CODE REVIEW

**Comment by marusak** [comment]
I like your current motd :)

File: test/verify/check-system-info [hunk 1]
```diff
@@ -355,6 +355,12 @@ class TestSystemInfo(MachineCase):
         b.wait_visible('#motd-box')
         b.wait_text('#motd', "Hello again")
 
+        b.click("#motd-box-edit")
+        b.set_input_text("#motd-box-edit-modal textarea", "Hello cockpit team")
+        b.click("#motd-box-edit-modal button.pf-m-primary")
+        b.wait_not_present("motd-box-edit-modal")
+        b.wait_text('#motd', "Hello cockpit team")
```
Comment (marusak) [inline test/verify/check-system-info:362]: Can you please check here that `/etc/motd` also contains this new text and it hasn't just changed in cockpit?

File: pkg/systemd/overview-cards/motdCard.jsx [hunk 2]
```diff
@@ -49,14 +54,60 @@ export class MotdCard extends React.Component {
...
+                       title={ <pre id="motd">{this.state.motdText}</pre> }
+                       actionClose={<>
+                           {superuser.allowed &&
+                           <Button variant="plain"
+                                   id="motd-box-edit"
+                                   onClick={() => this.setState({
+                                       motdEditMode: true,
+                                       motdTextEdited: this.state.motdText
+                                   })}
+                                   aria-label="Action">
```
Comment (marusak) [inline pkg/systemd/overview-cards/motdCard.jsx:77]: That is not a great label :) Can you put there something like `Edit motd` and don't forget to localize it as well.

**Review by marusak — CHANGES_REQUESTED** [review]
Awesome! Thanks :)
Just two tiny comments

File: pkg/systemd/overview-cards/motdCard.jsx [hunk 3]
```diff
@@ -49,14 +54,60 @@ export class MotdCard extends React.Component {
...
+                       title={ <pre id="motd">{this.state.motdText}</pre> }
+                       actionClose={<>
+                           {superuser.allowed &&
+                           <Button variant="plain"
+                                   id="motd-box-edit"
+                                   onClick={() => this.setState({
+                                       motdEditMode: true,
+                                       motdTextEdited: this.state.motdText
+                                   })}
+                                   aria-label={_("Edit motd")}
```
Comment (marusak) [inline pkg/systemd/overview-cards/motdCard.jsx:77]: needs `>`, see all builds are failing:
```
ERROR in ../pkg/systemd/overview-cards/motdCard.jsx
Module build failed (from ../node_modules/babel-loader/lib/index.js):
SyntaxError: /home/user/cockpit-project/bots/make-checkout-workdir/pkg/systemd/overview-cards/motdCard.jsx: Unexpected token (78:31)

  76 |                                    })}
  77 |                                    aria-label={_("Edit motd")}
> 78 |                                <EditIcon />
     |                                ^
  79 |                            </Button>}
  80 |                            <AlertActionCloseButton onClose={this.hideAlert} />
  81 |                        </>} />
```

**Review by marusak — CHANGES_REQUESTED** [review]

**Review by marusak — APPROVED** [review]
Thank you!

**Comment by garrett** [comment]
Just saw this.

If you don't already have an MOTD, what happens?

And does it handle /etc/motd.d/?

**Comment by marusak** [comment]
> If you don't already have an MOTD, what happens?

You cannot set it up. Also if you edit it and save empty message you won't be able to set it up to some other value. I thought about it, but that will require some additional design.

> And does it handle /etc/motd.d/?

We only show `/etc/motd` and not `/etc/motd.d/`.

**Pull request merged**